stan/math/rev/functor/coupled_ode_system.hpp

```diff
@@ -454,7 +460,10 @@ struct coupled_ode_system<F, var, var> {
     try {
       start_nested();
 
-      const vector<var> y_vars(z.begin(), z.begin() + N_);
+      vector<var> y_vars;
+      y_vars.reserve(N_);
+      for (std::size_t i = 0; i < N_; ++i)
+        y_vars.emplace_back(var(new vari(z[i], false)));
```

Comment:
Same.  

[optional from here on]

Which is making me think this should all be encapsulated in a construction function:

```
std::vector<var> to_vector_var(const std::vector<double>& x) {
  vector<var> y;
  y.reserve(x.size());
  for (double xi : x)
    y.emplace_back(new vari(xi, false));  
  return y;     
}
```

The  compiler would presumably be able to remove the copy on return.